test/tracerEq/test_bcs_2d.py

```diff
@@ -0,0 +1,161 @@
+from thetis import *
+import pytest
+import numpy as np
+
+
+def fourier_series_solution(mesh, time):
+    """
+    Consider a diffusion problem with a inhomogeneous Neumann condition and zero initial condition:
+       c_t = nu * c_xx, c_x(0, t) = diff_flux, c_x(l, t) = 0, c(x, 0) = 0
+
+    In order to solve it analytically, we decompose it into two diffusion problems:
+     - a diffusion problem with homogeneous Neumann conditions and a nonzero initial condition,
+           z_t = nu * z_xx, z_x(0, t) = 0, z_x(l, t) = 0, z(x, 0) = -ic;
+     - and a diffusion problem with homogeneous Neumann conditions and a nonzero source term,
+           w_t = nu * w_xx + s, w_x(0, t) = 0, w_x(l, t) = 0, w(x, 0) = 0.
+
+    Here ic = ic(x) is set as phi(x) = alpha(x) * diff_flux, where alpha(x) = -(l - x)**2 / (2l).
+    The source term s = s(x, t) is given by ic_t - nu*ic_xx = -nu*diff_flux/l
+
+    Solving the inhomogeneous problem amounts to summing the solutions of the homogeneous problems
+    and subtracting ic.
+
+    The exact solution takes the form of a Fourier series, which we truncate appropriately.
+    """
+    lx = 10
+    x, y = SpatialCoordinate(mesh)
+
+    # Diffusivity and diffusive flux BC to impose
+    nu = 0.1
```

Comment:
These values are now defined twice,  in the fourier call and in the run method. Just use the ones in the run method?